repose-aggregator/tests/performance-tests/roles/gatling/files/gatling.conf

```diff
@@ -0,0 +1,132 @@
+      #acceptAnyCertificate = true                     # When set to true, doesn't validate SSL certificates
+      #httpClientCodecMaxInitialLineLength = 4096      # Maximum length of the initial line of the response (e.g. "HTTP/1.0 200 OK")
+      #httpClientCodecMaxHeaderSize = 8192             # Maximum size, in bytes, of each request's headers
+      #httpClientCodecMaxChunkSize = 8192              # Maximum length of the content or each chunk
+      #webSocketMaxFrameSize = 10240000                # Maximum frame payload size
+      #sslEnabledProtocols = [TLSv1.2, TLSv1.1, TLSv1] # Array of enabled protocols for HTTPS, if empty use the JDK defaults
+      #sslEnabledCipherSuites = []                     # Array of enabled cipher suites for HTTPS, if empty use the AHC defaults
+      #sslSessionCacheSize = 0                         # SSLSession cache size, set to 0 to use JDK's default
+      #sslSessionTimeout = 0                           # SSLSession timeout in seconds, set to 0 to use JDK's default (24h)
+      #useOpenSsl = false                              # if OpenSSL should be used instead of JSSE (requires tcnative jar)
+      #useNativeTransport = false                      # if native transport should be used instead of Java NIO (requires netty-transport-native-epoll, currently Linux only)
+      #tcpNoDelay = true
+      #soReuseAddress = false
+      #soLinger = -1
+      #soSndBuf = -1
+      #soRcvBuf = -1
+      #allocator = "pooled"                            # switch to unpooled for unpooled ByteBufAllocator
+      #maxThreadLocalCharBufferSize = 200000           # Netty's default is 16k
+    }
+    dns {
+      #queryTimeout = 5000                             # Timeout of each DNS query in millis
+      #maxQueriesPerResolve = 6                        # Maximum allowed number of DNS queries for a given name resolution
+    }
+  }
+  jms {
+    #acknowledgedMessagesBufferSize = 5000             # size of the buffer used to tracked acknowledged messages and protect against duplicate receives
+  }
+  data {
+    writers = [console, file, graphite] # The list of DataWriters to which Gatling write simulation data (currently supported : console, file, graphite, jdbc)
+    console {
+      #light = false                    # When set to true, displays a light version without detailed request stats
+    }
+    file {
+      #bufferSize = 8192                # FileDataWriter's internal data buffer size, in bytes
+    }
+    leak {
+      #noActivityTimeout = 30  # Period, in seconds, for which Gatling may have no activity before considering a leak may be happening
+    }
+    graphite {
+      #light = false                    # only send the all* stats
+      host = "influxdb.openrepose.org"  # The host where the Carbon server is located
+      port = 13002                      # The port to which the Carbon server listens to (2003 is default for plaintext, 2004 is default for pickle)
```

Comment:
I won't worry about this for now. @Mario-Lopez proposed using a Virtual Cloud Network instead to provide better isolation and security for our hosts. If time permits, I will look into the solution, instead.

The form that using Ansible Vault would take is that we encrypt the port either in the roles default variables files, a file alongside it, or as a task variable on the template task. We then use the `template` module to template and copy the Gatling configuration file to the hosts that will run Gatling. As part of that process, the decrypted port will be templated into the configuration file. Of course, when Gatling is run, the encryption password must be provided either on the command line (by passing a password file), or by setting an environment variable. There are a few ways to accomplish that, but we would likely use Puppet to store the Vault password (in eyaml) and pass the password to the Ansible master host (whether it be the Jenkins slaves themselves or other cloud servers).